### PR TITLE
Migrate to using Parlour for generation

### PR DESCRIPTION
**Pull request by AaronC81**
This currently supports everything except warning line numbers, which have been temporarily removed from output. (I'm not sure how they could be implemented in Parlour given the way it's built, and I'm also not sure how necessary they are since the definition associated with each warning is also shown.)

@connorshea - if you'd be able to give this a quick code review at some point, that'd be super useful! Thanks

### CODE REVIEW

File: rbi/sord.rbi [hunk 1]
```diff
@@ -65,7 +65,7 @@ module Sord
         kind: Symbol,
         msg: String,
         item: YARD::CodeObjects::Base,
-        indent_level: Integer
+        indent_level: Integer,
```
Comment (connorshea) [inline rbi/sord.rbi]: Parlour (or Sord?) outputs params with commas on every line when multi-line. It's technically valid, but I don't like it.

File: lib/sord/rbi_generator.rb [hunk 2]
```diff
@@ -176,106 +111,100 @@ def add_methods(item, indent_level)
...
-
-        add_signature(sig_params_list, returns, indent_level)
+        parlour_params = parameter_names_and_defaults_to_tags
+          .zip(parameter_types)
+          .map do |((name, default), _), type|
+            Parlour::RbiGenerator::Parameter.new(name.to_s, type: type, default: default)
+          end
 
-        rbi_contents << "#{'  ' * (indent_level + 1)}def #{prefix}#{meth.name}(#{parameter_list}); end"
+        @current_object.create_method(meth.name.to_s, parlour_params, returns, class_method: meth.scope == :class)
```
Comment (connorshea) [inline lib/sord/rbi_generator.rb]: Perhaps a bit late to suggest this - and not really a problem with Sord - but the parameters should really use named parameters. Right now it's not super clear what's being passed in/why.

e.g.:
```ruby
@current_object.create_method(method_name: meth.name.to_s, params: parlour_params, returns: returns, class_method: meth.scope == :class)
```

**Comment by connorshea** [comment]
Also, you haven't pushed the latest master to the parlour repo (and also don't have any tags in the repo)

File: rbi/sord.rbi (hunk 1 above)
Comment (AaronC81) [reply rbi/sord.rbi]: I have mixed feelings about this too, to be honest - I might alter this on the Parlour side.

File: lib/sord/rbi_generator.rb (hunk 2 above)
Comment (AaronC81) [reply lib/sord/rbi_generator.rb]: This certainly helps readability, and if library changes are going to be made to Parlour, they may as well be made early on. I'll work on this!

**Comment by AaronC81** [comment]
Whoops, sorry - pushed now, with tags

File: rbi/sord.rbi (hunk 1 above)
Comment (connorshea) [reply rbi/sord.rbi]: Opened a PR to fix this :)

**Pull request merged**